### PR TITLE
[15.0][MIG] sale_coupon_commercial_partner_applicability: Renamed to coupon_commercial_partner_applicability + Migration to 15.0

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
Renamed to coupon_commercial_partner_applicability + Migration to 15.0

cc @Tecnativa TT43554

@chienandalu @stefan-tecnativa please review

### CODE REVIEW

File: coupon_commercial_partner_applicability/models/coupon_coupon.py [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
-    _inherit = "sale.coupon"
+class Coupon(models.Model):
+    _inherit = "coupon.coupon"
 
-    def _check_coupon_code(self, order):
+    def _check_coupon_code(self, order_date, partner_id, **kwargs):
         """All the partners in the commercial entity can use their common coupons"""
-        message = super()._check_coupon_code(order)
+        message = super()._check_coupon_code(order_date, partner_id, **kwargs)
+        order = kwargs.get("order")
```
Comment (chienandalu) [inline coupon_commercial_partner_applicability/models/coupon_coupon.py]: There's no need to get the order. Check how the base method does it and adapt to it

File: coupon_commercial_partner_applicability/tests/test_sale_coupon_apply_commercial_partner.py [hunk 2]
```diff
@@ -42,10 +42,19 @@ def setUpClass(cls):
 class TestSaleCouponApplyCommercialPartner(SaleCouponApplyCommercialPartnerCase):
     def test_sale_coupon_apply_commercial_partner(self):
         self.assertEqual(self.coupon.state, "new")
-        msg = self.coupon._check_coupon_code(self.sale)
+        order = self.sale
```
Comment (chienandalu) [inline coupon_commercial_partner_applicability/tests/test_sale_coupon_apply_commercial_partner.py]: This module doesn't depend on sale now, so you can't make a test with that model

**Comment by stefan-tecnativa** [comment]
Functionally tested 👍️

File: coupon_commercial_partner_applicability/tests/test_sale_coupon_apply_commercial_partner.py (hunk 2 above)
Comment (pilarvargas-tecnativa) [reply coupon_commercial_partner_applicability/tests/test_sale_coupon_apply_commercial_partner.py]: thanks :)

File: coupon_commercial_partner_applicability/models/coupon_coupon.py (hunk 1 above)
Comment (pilarvargas-tecnativa) [reply coupon_commercial_partner_applicability/models/coupon_coupon.py]: done, thanks :)

**Review by chienandalu — APPROVED** [review]
Thanks :)

**Comment by chienandalu** [comment]
> Functionally tested +1

There's no review

**Review by stefan-tecnativa — APPROVED** [review]

**Pull request merged**